### ISSUE TITLE
[Bug] Reset itself on restart in GrapheneOS (Build 2024111800)

### ISSUE THREAD

**Issue by sdazero** [zyrouge/symphony#678]
### Description

When restarting the player, it does not find the added files at first, then loads them seconds later. Playlists, favourites, etc. are then empty. It seems like a reset. The music folders are still added.

### Steps to Reproduce

Happens on every restart.

### Expected Behavior

Loading files, favorites, playlist as usual.

### Screenshots

![Screenshot_20241122-195447](https://github.com/user-attachments/assets/d8cc11ba-d06a-4e37-be2c-9374d3e8a2d0)

### Device

Google Pixel 8 Pro

### OS

Android 15

### Version

2024111800

### Contribution Guidelines

- [X] I agree to follow the [Contribution Guidelines](https://github.com/zyrouge/symphony/wiki/Contributions-Guidelines#issues).

**Comment by zyrouge** [comment]
Could you attach a clip of it? The app should load all the songs in less than a second. Also, there is no scoped storage permission in Android.

**Comment by sdazero** [comment]
Sure. It seems like its doing a refresh every restart. 

https://github.com/user-attachments/assets/8034dced-eebb-4977-8617-3491a012c7ff

**Comment by sdazero** [comment]
The Music folders contains 21GB Files.

**Comment by sdazero** [comment]
I have described one thing incorrectly. The favourites are still there. However, the player does a complete refresh every time it is started.

**Comment by speedit43** [comment]
I have this problem too but only when my storage runs low and I think that maybe to save space Graphene delete the cache

**Comment by sdazero** [comment]
> I have this problem too but only when my storage runs low and I think that maybe to save space Graphene delete the cache

I have checked and have 113GB free Disk Space actually.

**Comment by zyrouge** [comment]
> The Music folders contains 21GB Files.

This could be the issue. How many files are there? The scan speed depends on the number of files, depth the folder and cache hit/miss. My library has around 600 songs and it loads them under a second. It could be possible that cache gets invalidated for your somehow.

**Comment by sdazero** [comment]
I know this is not really an Argument but no other player tested (powetamp, simple music, metro, ... has this kind of Problem. You Start the player and all tracks Art immediately there. When I remember the time right this behaviour changed from Android 14 to 15. In 14 everything was fine.

**Comment by Holbormon** [comment]
Also having the issue on latest Graphene OS, Pixel 8. Folders are reloaded on app start, however for me it's more of a minor inconvenience, because thankfully the scan takes <1s, despite a comparable library size (31GB, ~2000 songs).

**Comment by Holbormon** [comment]
As a suggested workaround, could the app maybe not reset the track index between scans ? Currently, it feels like however long the scan takes, it starts from scratch every time. Instead, how about updating the index incrementally with the detected differences ? That way, the user would not notice however long the app takes to scan for new tracks on app restart.

**Comment by zyrouge** [comment]
> As a suggested workaround, could the app maybe not reset the track index between scans ? Currently, it feels like however long the scan takes, it starts from scratch every time. Instead, how about caching the library index and updating it incrementally with the detected differences ? That way, the user would not notice however long the app takes to scan for new tracks on app restart, and the app would remain usable (albeit not reflecting the very latest changes) in the meantime.

Well we already do that. Except the cache is only used after validation. Maybe this issue is specific to Android 15?

**Comment by sdazero** [comment]
Possible. Can you test it?

**Comment by zyrouge** [comment]
> Possible. Can you test it?

Unfortunately, I do not have a device running Android 15.

**Comment by sdazero** [comment]
You can test it with an emulator.
https://developer.android.com/studio?hl=en

**Comment by Holbormon** [comment]
> Well we already do that. Except the cache is only used after validation. Maybe this issue is specific to Android 15?

Not sure if this is related, but after trying other players to compare, it seems that the ones that work on my setup (Graphene, Android 15) are using a different way to fetch music files. They are asking for the "music and audio" permission, instead of the generic storage. A long shot perhaps but at least something to look into

**Comment by zyrouge** [comment]
Yup, the "music and audio" permission does not allow reading storage spaces such as SD card which can be easily done using scoped storage. But, MediaStore does caching on it's own but since we don't rely on it, we have to do it ourselves. Symphony cannot know the file changes in realtime (when the app is closed), so it does during the startup which seems to be causing this issue for you. Especially in this case, the cache will be invalidated when metadata such as modified time change when fetching from scoped storage.
